10.4/js/script.js

```diff
@@ -0,0 +1,141 @@
+// maps
+window.initMap = function () {
+  var map = new google.maps.Map(document.getElementById('map'), {
+    zoom: 11,
+    center: slides[0].coords
+  });
+  var markers = [];
+  var currentSlide;
+  // add markers
+  for (var i = 0; i < slides.length; i++) {
+    markers[i] = new google.maps.Marker({
+      position: slides[i].coords,
+      map: map
+    });
+  }
+
+  markers[0].addListener('click', function(){
+    flkty.select(0);
+  });
+  markers[1].addListener('click', function(){
+    flkty.select(1);
+  });
+  markers[2].addListener('click', function(){
+    flkty.select(2);
+  });
+  //etap 2
+  // wersja nie płynna działa
+  flkty.on('change', function(index){
+    map.panTo(markers[index].position);
+    map.setZoom(16);
+  })
+  // wersja płynna zawiesza przeglądarke
+  // flkty.on('change', function(index){
+  //   console.log(markers[index]);
+  //   smoothPanAndZoom(map, 16, markers[index].position);
```

Comment:
zeby to dzialalo plynnie, nie podawaj koordynatow z obiektu markera, tylko przekazuj obiekt z koordynatami bezposrenio z Twojej tablicy z html'a